CMakeLists.txt

```diff
@@ -8,16 +8,22 @@ if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif ()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
 if(NOT BUILD_TYPE)
     set(BUILD_TYPE Shared)
 endif ()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
 if(BUILD_TYPE STREQUAL Static)
     set(ENABLE_STATIC ON CACHE BOOL "" FORCE)
 elseif (BUILD_TYPE STREQUAL Shared)
     set(ENABLE_STATIC OFF CACHE BOOL "" FORCE)
 endif ()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
```

Comment:
Just curious, why do we need to set this three times?

And would this fit better inside `set_common_cxx_flags_for_redex`?

Reply:
I am still pretty new in this stuff, but I was wondering if we could use "set_common_cxx_flags_for_redex -pthread". Would this work ? I could try.

Reply:
@justinjhendrick I have made a new pull request with all the updates as of on this day.